### DISCUSSION TITLE
Mqtt broker

### DISCUSSION THREAD

**Discussion by rapzak** [micropython/micropython#12970] [General]
Hi,
Has anyone made a simple broker in ex. ESP32?

Maybe in size 5-10 units.

/Rapzak

**Comment by peterhinch** [comment]
This was discussed recently. It's not been done and I don't think the idea is appealing as a project. The simplest solution is a Raspberry Pi running Mosquitto. Even an ancient original Pi has ample power for the job.

**Reply by rapzak** [reply]
Hi, I want to do both at Client and broker on the ESP32, do let it both be a part of a bigger MQTT network, and form a smaller local MQTT network.... the smaller system has to continue to run if the main broker breaks down....
